### PR TITLE
fix(suite): graph worker file logic to avoid importing connect in worker

### PR DESCRIPTION
**Pull request by tomasklim**
## Description

- This PR addresses the `register` error observed in the development (localhost) console on the dashboard.
- The issue arose due to the addition of `@trezor/connect` into `@suite-common/wallet-config`. 
- The problem stems from `@trezor/connect` being bundled into the worker, which is problematic in itself. Although this issue should not break the app, it did so because:
  - Webpack added HMR code to one of the `@trezor/connect` files (`constants/errors`) - `__webpack_require__.$Refresh$.register(_c, "TypedError");`.
  - This addition is due to `...(isDev ? ['react-refresh/babel'] : []),` in `base.webpack.config.ts`.
  - If the `constants/errors` file of `@trezor/connect` is excluded in the rules, everything functions properly, because the HMR line was added just to one file in case of worker bundle.
  `exclude: [/node_modules/, /connect\/lib\/constants\/errors/]`
  - While a solution could involve separating webpack for the app and workers, we prefer not to maintain another config.
  - Connect is not required in the graph worker, but it was imported due to probably incorrect tree shaking.

I've resolved this by splitting the graph utils into multiple files. I'm aware that we generally avoid using index files, but in this case, I believe it makes sense to do so because of the possibility to add comments.

The graph logic was moved into a worker via this issue: https://github.com/trezor/trezor-suite/issues/1613. It's unclear whether it would still cause the app to freeze currently. 🤷‍♂️
By the way, there's an interesting screenshot of `Suite 0.0`, worth checking out. 

## Request for Ideas

- If you have any ideas on how to modify `webpack` to work even without splitting files, please share.

## Related Issue

Slack Discussion: https://satoshilabs.slack.com/archives/G019WLX2P7B/p1690821148119369

## Screenshots:

**Now:**

Screenshot 2023-08-01 at 19 39 16

**Before**:
Screenshot 2023-08-01 at 19 58 37
Screenshot 2023-08-01 at 19 58 45

`connect` is bundled into worker and, for some reason, HMR is included. To be able to see this error like this, you have to change `devtool: 'eval-source-map'` in `dev.webpack.config.ts` to `devtool: false`

Screenshot 2023-08-01 at 20 00 39

### CODE REVIEW

File: packages/suite/src/actions/suite/storageActions.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { getFormDraftKey } from '@suite-common/wallet-utils';
 import { FormDraftPrefixKeyValues } from '@suite-common/wallet-constants';
 import { STORAGE } from './constants';
 import { GraphData } from '../../types/wallet/graph';
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/storageActions.ts]:
```suggestion
import { GraphData } from 'src/types/wallet/graph';
```

File: packages/suite/src/utils/wallet/graph/utils.ts [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
 
 import {
-    AggregatedAccountHistory,
-    AggregatedDashboardHistory,
     CommonAggregatedHistory,
     GraphData,
     GraphRange,
     GraphScale,
-} from '../../types/wallet/graph';
+} from '../../../types/wallet/graph';
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/graph/utils.ts]:
```suggestion
} from 'src/types/wallet/graph';
```
and above

File: packages/suite/src/views/dashboard/components/PortfolioCard/components/DashboardGraph.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ import { Account } from 'src/types/wallet';
 import { variables, Button } from '@trezor/components';
 import { calcTicks, calcTicksFromData } from '@suite-common/suite-utils';
 import { AggregatedDashboardHistory } from '../../../../../types/wallet/graph';
```
Comment (szymonlesisz) [inline packages/suite/src/views/dashboard/components/PortfolioCard/components/DashboardGraph.tsx]:
```suggestion
import { AggregatedDashboardHistory } from 'src/types/wallet/graph';
```
same here

File: packages/suite/src/utils/wallet/graph/utilsWorker.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
+import BigNumber from 'bignumber.js';
+import { fromUnixTime, getUnixTime, startOfMonth } from 'date-fns';
+
+import {
+    AggregatedAccountHistory,
+    AggregatedDashboardHistory,
+    GraphData,
+} from '../../../types/wallet/graph';
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/graph/utilsWorker.ts]:
```suggestion
} from 'src/types/wallet/graph';
```

File: packages/suite/src/utils/wallet/graph/utilsShared.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import BigNumber from 'bignumber.js';
+import { AggregatedAccountHistory, AggregatedDashboardHistory } from '../../../types/wallet/graph';
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/graph/utilsShared.ts]:
```suggestion
import { AggregatedAccountHistory, AggregatedDashboardHistory } from 'src/types/wallet/graph';
```

File: packages/suite/src/actions/wallet/graphActions.ts [hunk 6]
```diff
@@ -21,7 +21,7 @@ import {
     enhanceBlockchainAccountHistory,
     accountGraphDataFilterFn,
     deviceGraphDataFilterFn,
-} from '../../utils/wallet/graphUtils';
+} from 'src/utils/wallet/graph';
 import { selectLocalCurrency } from '../../reducers/wallet/settingsReducer';
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/graphActions.ts]:
```suggestion
import { selectLocalCurrency } from 'src/reducers/wallet/settingsReducer';
```

and above `../../types/wallet/graph`

**Review by szymonlesisz — COMMENTED** [review]
nitpicks regarding imports, lets be consistent

**Review by matejkriz — APPROVED** [review]
agree with the import nitpics, otherwise it looks good to me 👍

File: packages/suite/src/actions/wallet/graphActions.ts (hunk 6 above)
Comment (tomasklim) [reply packages/suite/src/actions/wallet/graphActions.ts]: All imports from above and maybe some more fixed in 3f441332fca5581de32856deb5046442c3e4973d

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: desktop 23.9.0 (b1c5e1a274403a0e94696888c39c83743b69ae8a)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuiteDev/23.9.0 Chrome/112.0.5615.204 Electron/24.5.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T2T1 2.6.1 Universal (revision 4a6420c5dd038c06a6641331995c7b22cf4c699c)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**